test/modules/standard/jama/TestMatrix.chpl

```diff
@@ -0,0 +1,403 @@
+use Jama;
+use Math;
+
+/** TestMatrix tests the functionality of the Jama Matrix class and associated decompositions.
+<P>
+Run the test from the command line using
+<BLOCKQUOTE><PRE><CODE>
+ java Jama.test.TestMatrix 
+</CODE></PRE></BLOCKQUOTE>
+Detailed output is provided indicating the functionality being tested
+and whether the functionality is correctly implemented.   Exception handling
+is also tested.  
+<P>
+The test is designed to run to completion and give a summary of any implementation errors
+encountered. The final output should be:
+<BLOCKQUOTE><PRE><CODE>
+      TestMatrix completed.
+      Total errors reported: n1
+      Total warning reported: n2
+</CODE></PRE></BLOCKQUOTE>
+If the test does not run to completion, this indicates that there is a 
+substantial problem within the implementation that was not anticipated in the test design.  
+The stopping point should give an indication of where the problem exists.
+**/
+   proc main () {
+
+      var A,B,C,Z,O,I,R,S,X,SUB,M,T,SQ,DEF,SOL : Matrix;
+      // Uncomment this to test IO in a different locale.
+      // Locale.setDefault(Locale.GERMAN);
+      var errorCount=0;
+      var warningCount=0;
+      var tmp, s : real;
+
+      //double[] columnwise = {1.,2.,3.,4.,5.,6.,7.,8.,9.,10.,11.,12.};
```

Comment:
Definitely - i've added a reference to the original jama page and explained the code is a port from the original jama package.